SimPEG/electromagnetics/static/resistivity/sources.py

```diff
@@ -13,33 +13,87 @@ class BaseSrc(survey.BaseSrc):
     Base DC source
     """
 
-    current = properties.Float("amplitude of the source current", default=1.0)
+    current = properties.List(doc="amplitudes of the source currents", default=[1.0])
+    location = properties.List(
+        "location of the source electrodes",
+        survey.SourceLocationArray("location of electrode"),
+    )
 
     _q = None
 
-    def __init__(self, receiver_list, **kwargs):
+    def __init__(self, receiver_list, location, current=None, **kwargs):
         super(BaseSrc, self).__init__(receiver_list, **kwargs)
+        if type(location) is np.ndarray:
+            if location.ndim == 2:
+                location = list(location)
+            else:
+                location = [location]
+        if current is None:
+            current = self.current
+        if type(current) == float or type(current) == int:
+            current = [float(current)]
+        elif type(current) == np.ndarray:
+            if current.ndim == 2:
+                current = list(current)
+            else:
+                location = [location]
+        if len(current) != 1 and len(current) != len(location):
+            raise ValueError(
+                "Current must be constant or equal to the number of specified source locations."
+            )
+        if len(current) == 1:
+            current = np.repeat(current, len(location)).tolist()
+
+        self.current = current
+        self.location = location
 
     def eval(self, sim):
-        raise NotImplementedError
+        if self._q is not None:
+            return self._q
+        else:
+            if sim._formulation == "HJ":
+                inds = closestPoints(sim.mesh, self.location, gridLoc="CC")
+                self._q = np.zeros(sim.mesh.nC)
+                self._q[inds] = self.current
+            elif sim._formulation == "EB":
+                loc = np.row_stack(self.location)
+                cur = np.asarray(self.current)
+                interpolation_matrix = sim.mesh.get_interpolation_matrix(loc, locType='N').toarray()
+                q = np.sum(cur[:, np.newaxis] * interpolation_matrix, axis=0)
+                self._q = q
+            return self._q
 
     def evalDeriv(self, sim):
         return Zero()
 
 
+class Multipole(BaseSrc):
```

Comment:
@ngodber: Not sure what this is for. Please explain your intention and a use case.

Reply:
Please see below comment. Its generally useful modelling engineering scenarios where you are primarily interested in modelling distribution of currents.